### ISSUE TITLE
wrong errors and warnings

### ISSUE THREAD

**Issue by aalowlevel** [rust-lang/rust-analyzer#13276]
I could not fix this problem. it's very annoying. When i do "save file", this happens.
Edit: select end of line sequence -> LF does not fix this. it's not about wrong positions. it's just showing more errors and warnings that actually don't exist on cargo check.

**rust-analyzer version**: 0.3.1212-standalone (187bee0bb 2022-09-19)
**rustc version**: rustc 1.64.0 (a55dd71d5 2022-09-19)

**relevant settings**:

VS Code: Version: 1.71.2 (user setup)
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:03:37.738Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

![image](https://user-images.githubusercontent.com/66054880/191860086-6dcbf6b3-6359-4ba9-b9f7-4e78605d958e.png)

**Comment by bjorn3** [comment]
This warning you show does exist, but it is shown at the wrong location. It should be shown a couple lines lower. Did you edit the code between saving and taking this screenshot?

**Comment by seanandyrush** [comment]
no. when i do cargo check, those warnings and errors do not exist. (appmode is used)

**Comment by seanandyrush** [comment]
i took this screenshot after saving file (ctrl+s)

**Comment by seanandyrush** [comment]
and cargo check is different

**Comment by seanandyrush** [comment]
I am using cargo watch

**Comment by seanandyrush** [comment]
![image](https://user-images.githubusercontent.com/66054880/191997314-1f478516-5a66-46ba-9cd5-e72273380007.png)

**Comment by seanandyrush** [comment]
this is an uncomfortable situation that happens to me all the time.
![image](https://user-images.githubusercontent.com/66054880/191999491-aa0a9011-8a12-4400-b98d-d2e2d053d8a2.png)

**Comment by seanandyrush** [comment]
![image](https://user-images.githubusercontent.com/66054880/192000608-570aa06e-5149-4b71-ba8c-0e95915f3f41.png)
all files are saved.

**Comment by seanandyrush** [comment]
![image](https://user-images.githubusercontent.com/66054880/192006055-455de5bd-cb75-49f5-9d8c-9ef98b50718d.png)
![image](https://user-images.githubusercontent.com/66054880/192006103-09cc8407-85ff-4d95-8f09-a41b5438c3ef.png)

**Comment by seanandyrush** [comment]
![image](https://user-images.githubusercontent.com/66054880/192012461-a8d74315-8f92-4cf3-9f45-5ab71d365a7a.png)
looks like it's still showing old errors, while program is running.

**Comment by seanandyrush** [comment]
if i do cargo clean, it looks like fixed. however, back to the same problem after about 5-6 file saves.

**Comment by seanandyrush** [comment]
![image](https://user-images.githubusercontent.com/66054880/192100108-2e7049ce-1967-4306-823c-27695c149782.png)

**Comment by ChayimFriedman2** [comment]
This could happen when you edit the code after saving, like bjorn3 said. Otherwise, without reproduction steps there is nothing we can do.
